zarr/util.py

```diff
@@ -19,10 +19,24 @@
 }
 
 
+# from
+# https://stackoverflow.com/questions/50916422/python-typeerror-object-of-type-int64-is-not-json-serializable/50916741
+class NpEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.integer):
+            return int(obj)
+        elif isinstance(obj, np.floating):
+            return float(obj)
```

Comment:
Rather than only supporting int/float, you could support arbitrary NumPy scalars with:
```suggestion
        if isinstance(obj, np.generic):
            return obj.item()
```

Reply:
Yeah another way we might do this is...

```suggestion
        if isinstance(obj, numbers.Integral):
            return int(obj)
        elif isinstance(obj, numbers.Real):
            return float(obj)
```

Reply:
The other numpy scalar types that come to mind (missed by the current checks) are `bool_`, `bytes_` and `str_`.